src/test/scala/io/iohk/ethereum/pos/ElectionManagerImplSpec.scala

```diff
@@ -58,4 +54,44 @@ class ElectionManagerImplSpec extends FlatSpec with Matchers with PropertyChecks
     electionManager.verifyIsLeader(knownStakeholders(1), slotNumber = 3) shouldEqual false
     electionManager.verifyIsLeader(knownStakeholders(2), slotNumber = 3) shouldEqual false
   }
+
+  "ElectionManagerImpl" should "choose the correct miner, given a change in the stakeholder list" in new TestSetup {
+
+    val electionManager: ElectionManager= ElectionManagerImpl(changingMinersConfig)
+
+    // First round stakeholders
+    electionManager.verifyIsLeader(knownStakeholders(0), slotNumber = 1) shouldEqual true
+    electionManager.verifyIsLeader(knownStakeholders(1), slotNumber = 2) shouldEqual true
+    electionManager.verifyIsLeader(knownStakeholders(2), slotNumber = 3) shouldEqual true
+
+    // Second round stakeholders
+    electionManager.verifyIsLeader(otherKnownStakeholders(1), slotNumber = 4) shouldEqual true
+    electionManager.verifyIsLeader(otherKnownStakeholders(0), slotNumber = 5) shouldEqual true
+    electionManager.verifyIsLeader(otherKnownStakeholders(1), slotNumber = 6) shouldEqual true
+  }
+
+  trait TestSetup {
+
+    def configFromMinerStakeholderMapping(slotStakeholdersMapping: Map[BigInt, Seq[Address]]): OuroborosConfig =
+      new OuroborosConfig {
+        override val slotMinerStakeholdersMapping: Map[BigInt, Seq[Address]] = slotStakeholdersMapping
+
+        //unused
+        override val slotDuration: FiniteDuration = 0.millis
+      }
+
+    val knownStakeholders: Seq[Address] = List(
+      Address(hexString = "0x01"),
+      Address(hexString = "0x10"),
+      Address(hexString = "0x11"))
+
+    val otherKnownStakeholders: Seq[Address] = List(Address(100), Address(200))
```

Comment:
For the sake of completeness, I would not use disjoint sets of stakeholders, that is, for instance I would keep `0x10` as a miner from slot 4 onwards. That way, we would cover the 3 possible cases: 1) new stakeholders, 2) removed stakeholders, 3) "lingering" stakeholders.